src/mantarray_desktop_app/data_analyzer.py

```diff
@@ -550,6 +598,50 @@ def _process_stim_packet(self, stim_packet: Dict[Any, Any]) -> None:
         outgoing_msg = {"data_type": "stimulation", "data_json": outgoing_data_json}
         self._board_queues[0][1].put_nowait(outgoing_msg)
 
+    def _start_mag_finding_analysis(self, content: Dict[str, Any]) -> None:
+        recordings = content.get("recordings")
+
+        mag_analysis_thread = ErrorCatchingThread(
+            target=_mag_finding_analysis_thread,
+            args=(recordings, self._mag_analysis_output_dir, self._mag_analysis_thread_list),
+        )
+
+        mag_analysis_thread.start()
+
+        self._mag_analysis_active_thread = {
+            "thread": mag_analysis_thread,
+            "recordings": recordings,
+            "output_dir": self._mag_analysis_output_dir,
+        }
+
+    def _check_mag_analysis_statuses(self) -> None:
+        if thread_dict := self._mag_analysis_active_thread:
+            thread = thread_dict.get("thread")
+            recordings = thread_dict.get("recordings")
+            output_dir = thread_dict.get("output_dir")
+
+            if thread.is_alive():
```

Comment:
Calling `join()` will block until the thread completes, so we want to change this to `if not thread.is_alive():` so that we know the thread is completed before joining it

Reply:
yeah this was a major typo. I had switched it to something ele and forgot that part